### DISCUSSION TITLE
Main function in C library not being called.

### DISCUSSION THREAD

**Discussion by kunle-e** [dart-lang/ffi#162] [General]
I made a very basic C library into a plugin for Flutter via Dart ffi. I'm noticing that the main function is not being invoked automatically. I can call the doit function within the plugin from Flutter and get results, but main () is not being called on load. Is this expected behavior?

`#include <stdlib.h>
#include <stdio.h>
#include <android/log.h>

int main(){
 __android_log_print(ANDROID_LOG_DEBUG, "Flutter:", "MAIN CALLED %s", "");
 return 9;
}

int doit(){
  __android_log_print(ANDROID_LOG_DEBUG, "flutter", "DOIT CALLED %s", "");
 return 1;
}`

**Comment by dcharkes** [comment]
Yes, this is expected behavior.

The Dart `main` is run. When loading a `DynamicLibrary` and looking up a symbol, the C `main` is not executed. This is the same behavior as when one writes a C program (with a `main`) and loads another C file compiled to a dynamic library (which also has a `main`), the `main` in the dynamic library is not executed on loading the dynamic library.

**Reply by kunle-e** [reply]
Yes, makes perfect sense. Thanks!
